tools/src/main/scala/scala/scalanative/codegen/Generate.scala

```diff
@@ -246,19 +247,50 @@ object Generate {
                  Type.Int,
                  Val.Int(meta.moduleArray.size))
 
-    def genObjectArrayId() = {
-      val objectArray =
+    private def tpe2arrayId(tpe: String): Int = {
+      val clazz =
         linked
-          .infos(Global.Top("scala.scalanative.runtime.ObjectArray"))
+          .infos(Global.Top(s"scala.scalanative.runtime.${tpe}Array"))
           .asInstanceOf[Class]
 
+      meta.ids(clazz)
+    }
+
+    def genObjectArrayId(): Unit = {
       buf += Defn.Var(Attrs.None,
                       objectArrayIdName,
                       Type.Int,
-                      Val.Int(meta.ids(objectArray)))
+                      Val.Int(tpe2arrayId("Object")))
     }
 
-    def genTraitDispatchTables() = {
+    def genArrayIds(): Unit = {
+      val tpes = Seq("Unit",
+                     "Boolean",
+                     "Char",
+                     "Byte",
+                     "Short",
+                     "Int",
+                     "Long",
+                     "Float",
+                     "Double",
+                     "Object")
+      val ids = tpes.map(tpe2arrayId).sorted
+
+      // all the arrays have a common superclass, therefore their ids are consecutive
+      val min = ids.head
+      val max = ids.last
+      if (ids != (min to max)) {
+        throw new Exception(
+          s"Ids for all known arrays ($tpes) are not consecutive!")
+      }
+
+      buf += Defn.Var(Attrs.None, arrayIdsMinName, Type.Int, Val.Int(min))
+
+      buf += Defn.Var(Attrs.None, arrayIdsMaxName, Type.Int, Val.Int(max))
+
+    }
```

Comment:
In current implementation object arrays are indeed all going to get a contiguous sequence of class ids, but please add an assert to verify this here.

Reply:
It is already done in lines 282-285
```scala
      if (ids != (min to max)) {
        throw new Exception(
          s"Ids for all known arrays ($tpes) are not consecutive!")
      }
```

Reply:
You're right, missed that one.